### PR TITLE
PLAT-24187 move image to video conversion to concat job

### PR DESCRIPTION
**Pull request by salma2222**

### CODE REVIEW

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php [hunk 1]
```diff
@@ -356,11 +361,8 @@ public function startMultiClipConcatBatchJob($resources, $dbEntry, $clipEntry)
 	 * @param array $tempEntriesIds
 	 * @param string $destEntryId
 	 */
-	public function addMultiClipTrackEntries($sourceEntryIds, $multiTempEntryId, $tempEntriesIds, $destEntryId)
+	public function addMultiClipTrackEntries($sourceEntryIds, $multiTempEntryId, $destEntryId)
 	{
-		$description = "source entry Ids : [" . implode(",", $sourceEntryIds) . "] template entry ids: [" . implode(",",$tempEntriesIds) . "].";
```
[the remark this answers is not in the document]
Comment (salma2222) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php:361]: if we have more than 3 entries the text in admin console is cut, 
suggestion: we can show the temp entry Ids only instead

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php [hunk 2]
```diff
@@ -561,21 +563,63 @@ protected function handleClipConcatParentJob($batchJob)
 				}
 				else
 				{
-					$this->addClipJobsFromBatchJob($batchJob, $jobData);
+					$imageToVideo = $this->getJobDataConversionParams($jobData, self::IMAGE_TO_VIDEO);
+					if($imageToVideo)
```
[the remark this answers is not in the document]
Comment (salma2222) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php:567]: discussed offline

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php [hunk 3]
```diff
@@ -585,77 +629,113 @@ public function getCalculatedConversionParams($mediaInfoObjs, $conversionProfile
...
 
-		foreach ($mediaInfoObjs as $mediaInfoObj)
+		foreach ($resourcesData as $resourceData)
 		{
-			if(!$mediaInfoObj)
+			$entryId = $resourceData[self::SOURCE_ENTRY] ? $resourceData[self::SOURCE_ENTRY]->getId() : null;
+
+			/** @var $mediaInfoObj mediaInfo*/
+			$mediaInfoObj = $resourceData[self::MEDIA_INFO_OBJECT];
+			if(!($mediaInfoObj instanceof mediaInfo))
```
[the remark this answers is not in the document]
Comment (salma2222) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php:638]: unnecessary, it is included in !($mediaInfoObj instanceof mediaInfo) 
 null instanceof mediaInfo => false

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (inbal-ben-david) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php:361]: better have temp then nothing

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php [hunk 4]
```diff
@@ -585,77 +629,113 @@ public function getCalculatedConversionParams($mediaInfoObjs, $conversionProfile
...
+
+	protected function updateKeyFrequency(&$array, $key, $frequency)
+	{
+		$array["$key"] = isset($array["$key"]) ? $array["$key"] + $frequency : $frequency;
 	}
 
 	protected function decideAudioSamplingRate(array $allAudioSampleRates)
 	{
-		return min($allAudioSampleRates);
+		return 44100;
```
[the remark this answers is not in the document]
Comment (salma2222) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php]: Done

File: alpha/apps/kaltura/lib/clipconcat/kClipManager.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (salma2222) [reply alpha/apps/kaltura/lib/clipconcat/kClipManager.php:361]: added

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**